lib/Hakyll/Core/Rules.hs

```diff
@@ -160,19 +335,117 @@ version v rules = do
+to @posts\/hakyll.html@.
+
+__Compile Markdown to HTML and embed it in a template__
+
+> -- Select all Markdown files in 'posts' directory
+> match "posts/**.md" $ do
+>     route $ setExtension "html"
+>     compile $
+>         pandocCompiler >>=
+>           loadAndApplyTemplate "templates/post.html" defaultContext
+>
+> -- To Hakyll templates are just plain files that have to be processed
+> -- and placed into the store like any other file (but without routing).
+> -- e.g. file on disk: 'templates/post.html'
+> match "templates/*" $ compile templateBodyCompiler
+Note how a Markdown post that is compiled to HTML using
+'Hakyll.Web.Pandoc.pandocCompiler' in a first step and then embedded into
+a HTMl 'Hakyll.Web.Template.Template' in a second step by using
+'Hakyll.Web.Template.loadAndApplyTemplate'.
+We can use templates to to control the design and layout of a webpage.
+A template may look as follows:
+
+> <h1>$title$</h1>
+> $body$
+See "Hakyll.Web.Template" to see examples of the templating syntax.
+-}
+compile :: (Binary a, Typeable a, Writable a) => Compiler (Item a) -- ^ How to transform content
+                                              -> Rules ()          -- ^ Result
 compile compiler = Rules $ modify $ \s ->
     s {rulesCompiler = Just (fmap SomeItem compiler)}
 
 
 --------------------------------------------------------------------------------
--- | Add a route.
---
--- This adds a route for all items matching the current pattern.
-route :: Routes -> Rules ()
+{- | Add (or replace) routing in the current 'Rules' value.
+__This functions controls if and where the compiled results are written out__
+(use one of the 'match' functions to control what content is processed and
+'compile' to control how).
+See 'Hakyll.Core.Routes.Routes' and 'Hakyll.Core.Identifier.Identifier' for
+details on how output filepaths are computed.
+
+Hint:
+__If there's no route attached to a rule, the compilation result is not written out__.
+However, the compilation result saved to the 'Hakyll.Core.Store.Store'
+and can be loaded and used within another rule. This behavior is needed,
+for example, for templates.
+
+=== __Examples__
+__Rules with and without routing__
+
+> -- e.g. file on disk: 'templates/post.html'
+>
+> -- Rule 1 (without routing)
+> match "templates/*" $ do
+>     -- compilation result saved to store with implicit identifier, e.g. 'templates/post.html'
+>     compile templateCompiler
+>
+> -- Rule 2 (with routing)
+> match "posts/**.md" $ do
+>     route $ setExtension "html"
+>     compile $ do
+>        -- load compiled result of other rule with explicit identifier.
+>        postTemplate <- loadBody "templates/post.html"
+>        pandocCompiler >>= applyTemplate postTemplate defaultContext
+Note that we don't set a route in the first rule to avoid writing out our
+compiled templates (a webpage containing raw
+'Hakyll.Web.Template.templateCompiler' templates makes rarely sense after all).
```

Comment:
Not sure what you meant here. Perhaps rewrite it this way?

```suggestion
'Hakyll.Web.Template.templateCompiler' templates rarely makes sense, after all).
```

Reply:
I wanted to explain that it doesn't makes sense to render raw templates to an output file; that's why there's no routing part in the first rule. Maybe just leave the text inside parenthesis out completely?

Reply:
yeah, let's just remove it.